test/gtest/ucp/test_ucp_peer_failure.cc

```diff
@@ -570,3 +573,167 @@ UCS_TEST_P(test_ucp_peer_failure_keepalive, kill_receiver,
+
+        if (std::string(proto_name) == "rndv/put/zcopy") {
+            if (stage == UCP_PROTO_STAGE_START) {
+                if (!self->is_ep_closed) {
+                    scoped_log_handler slh(wrap_errors_logger);
+                    self->receiver().close_all_eps(*self, 0,
+                                                   UCP_EP_CLOSE_FLAG_FORCE);
+                    self->replace_ep_put_zcopy(req->send.ep);
+                    self->is_ep_closed = true;
+                }
+            } else {
+                EXPECT_TRUE(false) << "Unexpected proto stage";
+            }
+        }
+
+        ucs_status_t status = self->m_progress_storage[proto][stage](uct_req);
+
+        if (self->is_ep_closed) {
+            for (auto &elem : self->m_sender_uct_ops) {
+                elem.first->ops = elem.second;
+            }
+            self->protos_restore_progress();
+        }
+
+        return status;
+    }
+
+    void rndv_put_abort_test()
+    {
+        /* Do the first send-receive loop to establish the connection
+         * and check that connection works.
+         */
+        ucp_request_param_t req_param{};
+        auto sreq = ucp_tag_send_nbx(sender().ep(), &m_send_buf.front(),
+                                     m_send_buf.size(), 0, &req_param);
+        auto rreq = ucp_tag_recv_nbx(receiver().worker(), &m_recv_buf.front(),
+                                     m_recv_buf.size(), 0, 0, &req_param);
+        requests_wait({sreq, rreq});
+
+        /* Replace progress functions for all protocols and do another
+         * send-receive loop to close the ep during first rndv/put/zcopy call
+         * and replace put_zcopy to function that returns error. This should
+         * imitate network error.
+         */
+        protos_replace_progress();
+
+        {
+            scoped_log_handler slh(wrap_errors_logger);
+            sreq = ucp_tag_send_nbx(sender().ep(), &m_send_buf.front(),
+                                        m_send_buf.size(), 0, &req_param);
+            rreq = ucp_tag_recv_nbx(receiver().worker(), &m_recv_buf.front(),
+                                        m_recv_buf.size(), 0, 0, &req_param);
+
+            request_wait(rreq);
+            auto send_status = request_wait(sreq);
+            ASSERT_EQ(send_status, UCS_ERR_UNSUPPORTED);
+        }
+
+        protos_restore_progress();
+
+        /* Check that progress was called and ep was closed. */
+        ASSERT_EQ(is_ep_closed, true);
+    }
+
+    virtual void cleanup()
+    {
+        test_ucp_peer_failure::cleanup();
+
+        self = nullptr;
+    }
+
+protected:
+    using progress_array =
+            std::array<uct_pending_callback_t, UCP_PROTO_STAGE_LAST>;
+    using progress_hash_map =
+            std::unordered_map<const ucp_proto_t*, progress_array>;
+
+    constexpr static size_t m_buf_size{UCS_KBYTE};
+    std::array<char, m_buf_size> m_send_buf{}, m_recv_buf{};
+    bool is_ep_closed{false};
```

Comment:
m_is_ep_closed + alignment

Reply:
Renaming done, what's wrong with alignment? As I know we have code style rules only for assignment.

Reply:
struct fields and class members should aligned as well

Reply:
Done.